### DISCUSSION TITLE
Forecasting future with Orbit ML

### DISCUSSION THREAD

**Discussion by ichsan2895** [uber/orbit#615] [Q&A] [answered]
Hello, I recently test this code in my computer, and it works

```
import orbit
from orbit.utils.dataset import load_iclaims
from orbit.models.dlt import DLTFull, DLTMAP
from orbit.models import ets
from orbit.diagnostics.plot import plot_predicted_data, plot_predicted_components, plot_posterior_params
from orbit.constants.palette import QualitativePalette
from orbit.diagnostics.metrics import smape
from sklearn.preprocessing import MinMaxScaler

# log-transformed data
df = load_iclaims()
# train-test split
test_size = 52
train_df = df[:-test_size]
test_df = df[-test_size:]

dlt = DLTFull(
  response_col='claims', date_col='week',
  regressor_col=['trend.unemploy', 'trend.filling', 'trend.job'],
  seasonality=52,
)
dlt.fit(df=train_df)

# outcomes data frame
predicted_df = dlt.predict(df=test_df)

plot_predicted_data(
  training_actual_df=train_df, predicted_df=predicted_df,
  date_col=dlt.date_col, actual_col=dlt.response_col,
  test_actual_df=test_df
)
```

the `df` has 443 rows, the last date is **2018-06-24**. 

Now, how to predict the future (July and August 2018)? Sorry, I have googling and don't find any documentation about this.

Best regard, Ichsan

**Comment by edwinnglabs** [comment]
Thanks for checking in the example code.  That's a good question that we should have made this clear:
1. Forecastability: In this example, the regressors are actually contemporaneous with the response so it won't help in terms of a true forecast as you won't know the actual values for them in the future
2. 
2. Now-casting: however, this is still useful in terms of explaining what's happening the current and historical period as it can help you to do a what-if scenario analysis and answer question such as "what is the part can be explained by regressor X?" etc.
3. 
3. Using regressor for forecast: That said, you can still use regression part for forecasting in other situations, for example:
- use events and holidays as those information can be known prior to the forecast period
- use price, policy changes, treatment in forecasting demand of the product if you are the owner of those settings of the regressors

Hope this make sense to you @ichsan2895

**Comment by ichsan2895** [comment, marked as answer]
Thanks for reply, it is good if we predict each regressor (`trend.unemploy`, `trend.filling`, `trend.job`) independently (univariate) for future window, then use the result for regressor with `claims` trends?

**Reply by edwinnglabs** [reply]
> Thanks for reply, it is good if we predict each regressor (`trend.unemploy`, `trend.filling`, `trend.job`) independently (univariate) for future window, then use the result for regressor with `claims` trends?

There are some theories in time-series analysis that if the regressors are as predictable as the response itself, we may not need to worry about bringing in the regressors into the prediction equation as it can be predicted naturally as part of the response prediction process.  If we really think the regression component and response are happening in a multi-variate way, we should probably consider some multi-variate distribution instead, which we do consider them in our future roadmap.

**Reply by ichsan2895** [reply]
> > Thanks for reply, it is good if we predict each regressor (`trend.unemploy`, `trend.filling`, `trend.job`) independently (univariate) for future window, then use the result for regressor with `claims` trends?
> 
> There are some theories in time-series analysis that if the regressors are as predictable as the response itself, we may not need to worry about bringing in the regressors into the prediction equation as it can be predicted naturally as part of the response prediction process. If we really think the regression component and response are happening in a multi-variate way, we should probably consider some multi-variate distribution instead, which we do consider them in our future roadmap.

Thanks for sharing, I will close this thread
